website/content/docs/configuration/listener/shared.mdx

```diff
@@ -0,0 +1,88 @@
+---
+sidebar_label: Shared Listener Config
+description: |-
+  Shared config options for every listener type.
+---
+
+# Shared listener parameters
+
+There is config that is common between all listeners.
+
+## Client certificate header parameters
+
+- `x_forwarded_for_client_cert_header` `(string: "")` -
+  Specifies the header to be used for client certificate authentication. This is required
+  if [Client Certificate Auth](../../auth/cert.mdx) is used and your server is behind a reverse
+  proxy. It is recommended to have this set to the canonical `Client-Cert` header as defined in
+  [RFC 9440](https://www.rfc-editor.org/rfc/rfc9440.html). The header value should result in
+  base64 encoded DER at the end of processing.
+
+  :::warning
+
+  **Warning**: This should be used with `x_forwarded_for_authorized_addrs` set and
+  `x_forwarded_for_reject_not_present` and `x_forwarded_for_reject_not_authorized`
+  set to true if using the [`tcp`][tcp-listener] listener. Otherwise, if a direct request is made,
+  a public cert could be used without proper validation of the connection.
+  :::
+
+- `x_forwarded_for_client_cert_decoders` `(string array: [])` -
+  List of processors to handle `x_forwarded_for_client_cert_header`
+  These are processed in the order of the array.
+
+  Available values are
+  - URL - Urldecode the supplied value
+  - PEM - Convert a PEM certificate to base64 encoded DER
+  - RFC9440 - Use base64 encoded DER as defined in [RFC 9440 2.1][RFC 9440 2.1]
+
+  The result of this processing should end up with a base64 encoded DER certificate.
+  If the header value is already base64 encoded DER then no value needs to be set.
```

Comment:
RFC9440 is a great standard for forwarding client certificate information, but the reality is, that it is not used a lot (yet). In Kubernetes based environments that use a service mesh based on Envoy proxy, the [X-Forwarded-Client-Certificate](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-client-cert) header is what's being used.

If we intend this feature to be usable to authenticate at OpenBao using the inside a service mesh (wich we should IMO), then we must add support for the XFCC header as well.

A good library for parsing the XFCC header, which is also used by Istio Internally is:
https://github.com/alecholmes/xfccparser